src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -211,6 +215,28 @@ public void transform(KeyValue<LongWritable, GenericData.Record> input, Emitter<
 
   @Override
   public void onRunFinish(boolean succeeded, BatchSourceContext context) {
+    deleteGcsTemporaryDirectory();
+    deleteBigQueryTemporaryTable();
+  }
+
+  private void deleteBigQueryTemporaryTable() {
+    String temporaryTable = configuration.get(BigQueryConstants.CONFIG_TEMPORARY_TABLE_NAME);
+    if (temporaryTable != null) {
```

Comment:
This will always be non null

Reply:
@CuriousVini This will be null only in the case where we use the BigQuerySource for a Table without **filter** and table without **partition** configured. Therefore we had to add this check. Test case #1 from the comment below.

Reply:
We always set it in line 174 right?

Reply:
@CuriousVini in line 174 only the table name gets created. In the class `PartitionedBigQueryInputFormat` line 114 the temporary table gets created if `query != null` (this happens when we query a Table with filter or **partition** configured.

Reply:
so this check is for the string right? Meaning it will never be null since we always create it. Wondering why we need this check then? 
```
if (temporaryTable != null)

Reply:
@CuriousVini we apologize for the confusion, yes you are right. We do not need this check here. It does not make sense.